// @aws-sdk/client-transcribe/dist-es/commands/StartMedicalTranscriptionJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartMedicalTranscriptionJobRequest, StartMedicalTranscriptionJobResponse } from "../models/models_0";
import { deserializeAws_json1_1StartMedicalTranscriptionJobCommand, serializeAws_json1_1StartMedicalTranscriptionJobCommand, } from "../protocols/Aws_json1_1";
var StartMedicalTranscriptionJobCommand = (function (_super) {
    __extends(StartMedicalTranscriptionJobCommand, _super);
    function StartMedicalTranscriptionJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartMedicalTranscriptionJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "StartMedicalTranscriptionJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartMedicalTranscriptionJobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartMedicalTranscriptionJobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartMedicalTranscriptionJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StartMedicalTranscriptionJobCommand(input, context);
    };
    StartMedicalTranscriptionJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StartMedicalTranscriptionJobCommand(output, context);
    };
    return StartMedicalTranscriptionJobCommand;
}($Command));
export { StartMedicalTranscriptionJobCommand };
